tests/integration/test_materialize_mysql_database/materialize_with_ddl.py

```diff
@@ -461,3 +463,81 @@ def select_without_columns(clickhouse_node, mysql_node, service_name):
+    time.sleep(5)
+
+def drop_instance_mysql_connections(clickhouse_node, pm, action='DROP'):
+    pm._check_instance(clickhouse_node)
+    pm._add_rule({'source': clickhouse_node.ip_address, 'destination_port': 3306, 'action': action})
+    pm._add_rule({'destination': clickhouse_node.ip_address, 'source_port': 3306, 'action': action})
+    time.sleep(5)
+
+def network_partition_test(clickhouse_node, mysql_node, service_name):
+    mysql_node.query("CREATE DATABASE test_database;")
+    mysql_node.query("DROP DATABASE IF EXISTS test")
+    clickhouse_node.query("DROP DATABASE IF EXISTS test")
+    mysql_node.query("CREATE DATABASE test;")
+    mysql_node.query("CREATE TABLE test_database.test_table ( `id` int(11) NOT NULL, PRIMARY KEY (`id`) ) ENGINE=InnoDB;")
+
+    with PartitionManager() as pm:
+        clickhouse_node.query(
+            "CREATE DATABASE test_database ENGINE = MaterializeMySQL('{}:3306', 'test_database', 'root', 'clickhouse')".format(service_name))
+
+        drop_instance_mysql_connections(clickhouse_node, pm)
+
+        mysql_node.query('INSERT INTO test_database.test_table VALUES(1)')
+        check_query(clickhouse_node, "SHOW TABLES FROM test_database FORMAT TSV", "test_table\n")
+        check_query(clickhouse_node, "SELECT * FROM test_database.test_table", '')
+
+        with pytest.raises(QueryRuntimeException) as exception:
+            clickhouse_node.query(
+                "CREATE DATABASE test ENGINE = MaterializeMySQL('{}:3306', 'test', 'root', 'clickhouse')".format(service_name))
+
+        assert "Can't connect to MySQL server" in str(exception.value)
+
+        clickhouse_node.query("DETACH DATABASE test_database")
+
+        restore_instance_mysql_connections(clickhouse_node, pm)
+
+        clickhouse_node.query("ATTACH DATABASE test_database")
+        check_query(clickhouse_node, "SELECT * FROM test_database.test_table FORMAT TSV", '1\n')
+
+        clickhouse_node.query("DROP DATABASE test_database")
+        mysql_node.query("DROP DATABASE test_database")
+
+def mysql_kill_sync_thread_restore_test(clickhouse_node, mysql_node, service_name):
+    mysql_node.query("CREATE DATABASE test_database;")
+    mysql_node.query("CREATE TABLE test_database.test_table ( `id` int(11) NOT NULL, PRIMARY KEY (`id`) ) ENGINE=InnoDB;")
+    mysql_node.query("INSERT INTO test_database.test_table VALUES (1)")
+
+    clickhouse_node.query("CREATE DATABASE test_database ENGINE = MaterializeMySQL('{}:3306', 'test_database', 'root', 'clickhouse')".format(service_name))
+    check_query(clickhouse_node, "SELECT * FROM test_database.test_table FORMAT TSV", '1\n')
+
+    result = mysql_node.query_and_get_data("select id from information_schema.processlist where STATE='Master has sent all binlog to slave; waiting for more updates'")
+
+    for row in result:
+        row_result = {}
+        query = "kill " + str(row[0]) + ";"
+        mysql_node.query(query)
+
+    time.sleep(5)
```

Comment:
Tests with `sleep(...)` are usually flacky, but let's look at the result of flacky check.

Reply:
If I want to have a MySQL client writing data continuously and then close MySQL Server Docker during the writing process.  Next, stop the client and restart the MySQL Server Docker. Finally check the MySQL data rows and ClickHouse data rows.

Can this finish in integration testing? @tavplubix